src/frontend/src/pages/GanttPage/GanttChartSection.tsx

```diff
@@ -48,6 +45,20 @@ const GanttChartSection = ({
     setCurrentTask(undefined);
   };
 
+  const handleMouseMove = (e: MouseEvent) => {
+    setCursorPosition({ x: e.clientX, y: e.clientY });
+  };
+
+  useEffect(() => {
+    if (!isEditMode) {
+      window.addEventListener('mousemove', handleMouseMove);
+    }
+
+    return () => {
+      window.removeEventListener('mousemove', handleMouseMove);
+    };
+  }, [isEditMode]);
```

Comment:
is there a reason this needs to be in a useEffect, rather than just being an if statement in the component, since isEditMode is a prop and should cause a re-render whenever it changes anyways?

Reply:
When I did a if statement, it was still super laggy and basically froze the entire site and the useEffect is the only thing that fixed it

Reply:
hrmm, ok that makes sense, this should have the same end result, so it's not that deep